packages/xstate-react/src/useActor.ts

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect, useMemo, useRef, useCallback } from 'react';
+import { ActorRef, Sender, ActorRefLike } from './types';
+import { EventObject, Interpreter, Actor } from 'xstate';
+import { fromService } from './useService';
+
+function resolveActor(
+  actorLike:
+    | ActorRef<any, any>
+    | ActorRefLike<any, any>
+    | Interpreter<any, any>
+    | Actor
+): ActorRef<any, any> & { deferred?: boolean } {
+  if (actorLike instanceof Interpreter) {
+    return fromService(actorLike);
+  }
+
+  if (!('current' in actorLike)) {
+    return {
+      stop: () => {
+        /* do nothing */
+      },
+      ...actorLike,
+      subscribe: actorLike.subscribe as any, // TODO: fix (subscribe will work, just improperly typed)
+      name: actorLike.id!,
+      current: actorLike.state!
+    };
+  }
+
+  return actorLike;
+}
+
+export function useActor<TEvent extends EventObject, TEmitted = any>(
+  actorLike: ActorRefLike<TEvent, TEmitted> | Actor
+): [TEmitted, Sender<TEvent>] {
+  const actor = useMemo(() => resolveActor(actorLike), [actorLike]);
+  const deferredEventsRef = useRef<EventObject[]>([]);
+  const [current, setCurrent] = useState(actor.current);
+
+  const send: Sender<TEvent> = useCallback(
```

Comment:
does this work right now? I can't find anywhere in the code `.deferred` being switched to false

Reply:
oh, ok - i think I understand this now - the identity of a null actor is still replaced after we start a machine, but this time it has no deferred property at all

Reply:
Yep, that's correct.